app/code/Magento/User/Controller/Adminhtml/Auth/Forgotpassword.php

```diff
@@ -7,26 +7,45 @@
 namespace Magento\User\Controller\Adminhtml\Auth;
 
 use Magento\Security\Model\SecurityManager;
+use Magento\Framework\App\ObjectManager;
+use Magento\Backend\App\Action\Context;
+use Magento\User\Model\UserFactory;
+use Magento\User\Model\ResourceModel\User\CollectionFactory;
+use Magento\Framework\Validator\EmailAddress;
+use Magento\Security\Model\PasswordResetRequestEvent;
+use Magento\Framework\Exception\SecurityViolationException;
+use Magento\User\Controller\Adminhtml\Auth;
 
-class Forgotpassword extends \Magento\User\Controller\Adminhtml\Auth
+class Forgotpassword extends Auth
 {
     /**
      * @var SecurityManager
      */
     protected $securityManager;
+    
+    /**
+     * User model factory
+     *
+     * @var CollectionFactory
+     */
+    private $userCollectionFactory;
 
     /**
-     * @param \Magento\Backend\App\Action\Context $context
-     * @param \Magento\User\Model\UserFactory $userFactory
-     * @param \Magento\Security\Model\SecurityManager $securityManager
+     * @param Context $context
+     * @param UserFactory $userFactory
+     * @param SecurityManager $securityManager
+     * @param CollectionFactory $userCollectionFactory
      */
     public function __construct(
-        \Magento\Backend\App\Action\Context $context,
-        \Magento\User\Model\UserFactory $userFactory,
-        \Magento\Security\Model\SecurityManager $securityManager
+        Context $context,
+        UserFactory $userFactory,
+        SecurityManager $securityManager,
+        CollectionFactory $userCollectionFactory = null
     ) {
         parent::__construct($context, $userFactory);
         $this->securityManager = $securityManager;
+        $this->userCollectionFactory = $userCollectionFactory ?:
```

Comment:
Please do PR on 2.3-develop and remove the `?: ObjectManager::getInstance()->get(Data::class);` part

Reply:
This part was done correctly, no need to introduce backward incompatible changes if possible